### PR TITLE
clean up device crd e2e test

### PR DESCRIPTION
**Pull request by wackxu**
Signed-off-by: wackxu <xushiwei5@huawei.com>

**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

We want to eliminate insecure ways to access the apiserver and use kube config and client-go to access the kube apiserver.
This is required to run the kubeedge conformance test

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: tests/e2e/utils/common.go [hunk 1]
```diff
@@ -373,93 +327,75 @@ func newDeviceModelObject(protocolType string, updated bool) *v1alpha2.DeviceMod
...
+// CheckDeviceModelExists verify whether the contents of the device model matches with what is expected
+func CheckDeviceModelExists(deviceModels []v1alpha2.DeviceModel, expectedDeviceModel *v1alpha2.DeviceModel) error {
+	modelExists := false
+	for _, deviceModel := range deviceModels {
+		if expectedDeviceModel.ObjectMeta.Name == deviceModel.ObjectMeta.Name {
+			modelExists = true
+			if !reflect.DeepEqual(expectedDeviceModel.TypeMeta, deviceModel.TypeMeta) ||
+				expectedDeviceModel.ObjectMeta.Namespace != deviceModel.ObjectMeta.Namespace ||
+				!reflect.DeepEqual(expectedDeviceModel.Spec, deviceModel.Spec) {
+				return fmt.Errorf("the device model is not matching with what was expected")
```
Comment (gy95) [inline tests/e2e/utils/common.go:355]: add a break here, or return nil directly

File: tests/e2e/utils/common.go [hunk 2]
```diff
@@ -373,93 +327,75 @@ func newDeviceModelObject(protocolType string, updated bool) *v1alpha2.DeviceMod
...
+				!reflect.DeepEqual(expectedDevice.Spec, device.Spec) {
+				return fmt.Errorf("the device is not matching with what was expected")
+			}
+			twinExists := false
+			for _, expectedTwin := range expectedDevice.Status.Twins {
+				for _, twin := range device.Status.Twins {
+					if expectedTwin.PropertyName == twin.PropertyName {
+						twinExists = true
+						if !reflect.DeepEqual(expectedTwin.Desired, twin.Desired) {
+							return fmt.Errorf("Status twin " + twin.PropertyName + " not as expected")
```
Comment (gy95) [inline tests/e2e/utils/common.go:383]: add break

File: tests/e2e/utils/common.go [hunk 3]
```diff
@@ -373,93 +327,75 @@ func newDeviceModelObject(protocolType string, updated bool) *v1alpha2.DeviceMod
...
 		}
-		if !deviceExists {
-			return nil, fmt.Errorf("the requested device is not found")
-		}
 	}
-	return list.Items, nil
+
+	if !deviceExists {
+		return fmt.Errorf("the requested device is not found")
+	}
```
Comment (gy95) [inline tests/e2e/utils/common.go:396]: add a break

**Review by gy95 — COMMENTED** [review]
LGTM

**Review by vincentgoat — COMMENTED** [review]
lgtm. Replacing with `ClientSet` would be more secure.

**Pull request merged**